DOCUMENTATION.md

```diff
@@ -1,810 +1,807 @@
 
-#### Arguments
+_NEED EXAMPLE_
 
-<details>
-<summary>
-Click here to expand for more information.
-</summary>
+### Error codes
 
-##### `template_type`
-If omitted all templates are returned. Otherwise you can filter by:
+If the request is not successful, the client will return a `Notifications::Client::RequestError` containing the relevant error code:
 
-* `email`
-* `sms`
-* `letter`
+|error.code|error.message|How to fix|
+|:---|:---|:---|
+|`400`|`[{`<br>`"error": "ValidationError",`<br>`"message": "Template type is not one of [sms, email, letter]"`<br>`}]`|Contact the Notify team|
+
+## Generate a preview template
 
-</details>
+### Method
 
+This will generate a preview version of a template.
 
-## Generate a preview template
+```ruby
+template_preview: client.generate_template_preview(id, personalisation{})
+```
 
-#### Method
+The parameters in the personalisation argument must match the placeholder fields in the actual template. The API notification client will ignore any extra fields in the method.
 
-This will return the contents of a template with the placeholders replaced with the given personalisation.
+### Arguments
 
-<details>
-<summary>
-Click here to expand for more information.
-</summary>
+#### id (required)
 
+The ID of the template. You can find this by signing into GOV.UK Notify and going to the __Templates__ page.
 
 ```ruby
-templatePreview = client.generate_template_preview(id,
-                                                  personalisation: {
-                                                      name: "name",
-                                                      year: "2016",                      
-                                                    })
+id: 'f33517ff-2a88-4f6e-b855-c550268ce08a';
 ```
 
-</details>
-
+#### personalisation (required)
 
-#### Response
+If a template has placeholder fields for personalised information such as name or application date, you must provide their values in a hash. For example:
 
-<details>
-<summary>
-Click here to expand for more information.
-</summary>
-
-```Ruby
-template.id         # => uuid for the template
-template.version    # => version of the template
-template.body       # => content of the template
-template.subject    # => subject for email templates, will be empty for other template types
+```ruby
+personalisation: {
+  name: "John Smith",
+  ID: "300241",                      
+}
 ```
-Otherwise a `Notifications::Client::RequestError` is thrown.
 
-|`error.code`|`error.message`|
-|:---|:---|
-|`404`|`[{`<br>`"error": "NoResultFound",`<br>`"message": "No result found"`<br>`}]`|
-|`404`|`[{`<br>`"error": "ValidationError",`<br>`"message": "is is not a valid UUID"`<br>`}]`|
+You can leave out this argument if a template does not have any placeholder fields for personalised information.
+
+### Response
 
-</details>
+If the request to the client is successful, the client will return a `TemplatePreview`:
```

Comment:
For consistency, should probably say `Notifications::Client::TemplatePreview` instead of just `TemplatePreview`